File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/SimpleStorageCleaner.java
package com.example.alexeyglushkov.cachemanager;

import com.example.alexeyglushkov.streamlib.progress.ProgressUpdater;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.functions.Consumer;

/**
 * Created by alexeyglushkov on 10.10.15.
 */

public class SimpleStorageCleaner implements StorageCleaner {
    public SimpleStorageCleaner() {
    }

    @Override
    public void setProgressInfo(ProgressUpdater info) {
    }

    @Override
    public Completable clean(Storage storage) {
        RxStorageAdapter asyncStorage = new RxStorageAdapter(storage);
        return clean(asyncStorage);
    }

    @Override
    public Completable clean(RxStorage rxStorage) {
        return rxStorage.getEntries()
            .doOnSuccess(new Consumer<List<StorageEntry>>() {
                @Override
                public void accept(List<StorageEntry> entries){
                    for (StorageEntry entry : entries) {
                        StorageMetadata metadata = entry.getMetadata();

                        if (metadata != null && metadata.isExpired()) {
                            try {
                                entry.delete();

                            } catch (Exception e) {
                                e.printStackTrace();
                            }
                        }
                    }
                }
            })
            .ignoreElement();
    }
}

File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/SimpleStorageCleaner.kt
package com.example.alexeyglushkov.cachemanager

import com.example.alexeyglushkov.streamlib.progress.ProgressUpdater
import io.reactivex.Completable

/**
 * Created by alexeyglushkov on 10.10.15.
 */
class SimpleStorageCleaner : StorageCleaner {
    override fun setProgressInfo(info: ProgressUpdater?) {}
    override fun clean(storage: Storage): Completable {
        val asyncStorage = RxStorageAdapter(storage)
        return clean(asyncStorage)
    }

    override fun clean(storage: RxStorage): Completable {
        return storage.getEntries()
                .doOnSuccess { entries ->
                    for (entry in entries) {
                        val metadata = entry.metadata
                        if (metadata != null && metadata.isExpired()) {
                            try {
                                entry.delete()
                            } catch (e: Exception) {
                                e.printStackTrace()
                            }
                        }
                    }
                }
                .ignoreElement()
    }
}